ci-operator/step-registry/telco5g/origin/tests/telco5g-origin-tests-commands.sh

```diff
@@ -37,6 +35,13 @@ cd "$cnf_dir" || exit 1
 echo "running on branch ${CNF_BRANCH}"
 git clone -b "${CNF_BRANCH}" "${CNF_REPO}" cnf-features-deploy
 cd cnf-features-deploy
+if [[ "$T5CI_VERSION" == "4.15" ]]; then
+    echo "Updating all submodules for >=4.15 versions"
+    # git version 1.8 doesn't work well with forked repositories, requires a specific branch to be set
+    sed -i "s@https://github.com/openshift/metallb-operator.git@https://github.com/openshift/metallb-operator.git\n        branch = main@" .gitmodules
+    git submodule update --init --force --recursive --remote
+    git submodule foreach --recursive 'echo $path `git config --get remote.origin.url` `git rev-parse HEAD`' | grep -v Entering > ${ARTIFACT_DIR}/hashes.txt || true
+fi
```

Comment:
Looks good to me

I don't know the step completely but I tried to review the bash added in the PR.

sed command works locally
sed -i "s@https://github.com/openshift/metallb-operator.git@https://github.com/openshift/metallb-operator.git\n        branch = main@" .gitmodules

$ git diff
diff --git a/.gitmodules b/.gitmodules
index 5e2c3eb2..fa8876ed 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,6 +1,7 @@
 [submodule "cnf-tests/submodules/metallb-operator"]
        path = cnf-tests/submodules/metallb-operator
        url = https://github.com/openshift/metallb-operator.git
+        branch = main

the only thing I wonder is do we need || true command in line 43, as the redirection will always succeed, even if the hashes.txt file does not exist or is empty.

Reply:
It covers the grep part `grep -v Entering`